### ISSUE TITLE
Implement pre-build copy

### ISSUE THREAD

**Issue by Emilgardis** [cross-rs/cross#861]
### Checklist

- [X] I've looked through the [issues and pull requests](https://github.com/cross-rs/cross/issues?q=) for similar request
- [X] This feature could be solved with a [custom docker image](https://github.com/cross-rs/cross#custom-docker-images) (optional)

### Describe your request

From [reddit](https://www.reddit.com/r/rust/comments/vk2xfc/cross_v022_released/idn2x8e/)

Make it easy to integrate scripts to the pre-build hook.

**Comment by Alexhuszagh** [comment]
I'm not sure this is needed: I remember being initially very favorable of the idea of supporting other commands in the pre-build hooks or customizing the base image, but in retrospect it seems like the solution we implemented was much better.

Currently, we have:

```toml
[target.x86_64-unknown-linux-gnu]
pre-build = [
    "dpkg --add-architecture arm64", 
    "apt-get update && apt-get install --assume-yes libssl-dev:arm64"
]
```

And my initial proposal would have been:

```toml
[target.x86_64-unknown-linux-gnu]
pre-build = [
    "RUN dpkg --add-architecture arm64", 
    "RUN apt-get update && apt-get install --assume-yes libssl-dev:arm64"
]
```

Which honestly in retrospect is not very legible, and a Dockerfile is better in that case. If we use a `copy`, will all copies be done before all runs? How will this affect cached builds if another step is added? What else do we need to implement? `ENV`? `ARG`? I think just using a `Dockerfile` or simple pre-build `RUN` instructions is enough.

**Comment by Emilgardis** [comment]
We only have one `RUN` inserted, each value in `pre-build` is separated by a newline and evaluated with `eval`

I think it'd be fine to include a way to call a script specifically, maybe that usecase is what should be supported and not a general "copy"

**Comment by Alexhuszagh** [comment]
> We only have one `RUN` inserted, each value in `pre-build` is separated by a newline and evaluated with `eval`
> 
> I think it'd be fine to include a way to call a script specifically, maybe that usecase is what should be supported and not a general "copy"

I like that better, and make it clear it would happen after the pre-build run commands, which I believe is best.

**Comment by Emilgardis** [comment]
maybe simply

```toml
[*]
pre-build = "./myscript.sh" # Note: not an array.
```
don't think we need to support pre-build via array and this new alternative

**Comment by Alexhuszagh** [comment]
That makes it super simple, and correct. I think that's the correct approach.

**Comment by Alexhuszagh** [comment]
This was closed in #910.
